### PR TITLE
Feature/token auction deploy

### PR DESCRIPTION
**Pull request by cducrest**

### CODE REVIEW

File: tools/auction-deploy/src/auction_deploy/core.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
-from typing import Dict, NamedTuple, Optional, Sequence
+from typing import Dict, NamedTuple, Optional, Sequence, Tuple
 
 from deploy_tools.deploy import (
     deploy_compiled_contract,
     increase_transaction_options_nonce,
     load_contracts_json,
     send_function_call_transaction,
 )
+from eth_tester.exceptions import TransactionFailed
```
Comment (berndbohmeier) [inline tools/auction-deploy/src/auction_deploy/core.py:9]: core should not import stuff from eth tester

File: tools/auction-deploy/src/auction_deploy/core.py [hunk 2]
```diff
@@ -152,6 +183,20 @@ def get_deployed_auction_contracts(
...
+def get_bid_token_address(web3, auction_address: str):
+    compiled_contracts = load_contracts_json(__name__)
+    auction_abi = compiled_contracts["TokenValidatorAuction"]["abi"]
+    auction = web3.eth.contract(address=auction_address, abi=auction_abi)
+    try:
+        return auction.functions.bidToken().call()
+    except BadFunctionCallOutput:
+        # Thrown by web3 when function does not exist on contract
+        return None
+    except TransactionFailed:
```
Comment (berndbohmeier) [inline tools/auction-deploy/src/auction_deploy/core.py:195]: How did you test the BadFunctionCallOutput if this is raised by the tester?
I dont like that we here handle the tester specific, I would hope we can somehow remove this.

File: tools/auction-deploy/tests/conftest.py [hunk 3]
```diff
@@ -1,17 +1,30 @@
...
+
+assert (
+    eth_tester.backends.pyevm.main.GENESIS_GAS_LIMIT < 8 * 10 ** 6
+)  # noqa: E402, E702 isort:skip
+eth_tester.backends.pyevm.main.GENESIS_GAS_LIMIT = (
+    8 * 10 ** 6
+)  # noqa: E402, E702 isort:skip
+
+
+import pytest  # noqa: E402, E702 isort:skip
```
Comment (berndbohmeier) [inline tools/auction-deploy/tests/conftest.py:17]: it is fine that the stuff above have the ignore linting stuff, but that here every line has it is a bit much. Please find a way to only have to exclude the eth_tester lines.

File: tools/bridge/pytest [hunk 4]
```diff
@@ -6,10 +6,10 @@ from gevent import monkey  # isort:skip
 
 monkey.patch_all()  # noqa: E402 isort:skip
 
-import re
-import sys
+import re  # noqa: E402, E702 isort:skip
```
Comment (berndbohmeier) [inline tools/bridge/pytest:9]: same as above.

File: tools/auction-deploy/src/auction_deploy/core.py (hunk 2 above)
Comment (cducrest) [reply tools/auction-deploy/src/auction_deploy/core.py:195]: I tested the BadFunctionCallOutput by deploying an auction on goerli and calling status on it

File: tools/auction-deploy/tests/conftest.py (hunk 3 above)
Comment (cducrest) [reply tools/auction-deploy/tests/conftest.py]: Took me quite some time to figure it out. My understanding is that the flake8 ignore for `E402` has to be on every line that is an import below a non import statement or that the non import statement has to be on the same line as an import statement (for some reason).

Tell me if you like the current version better.

**Pull request merged**